src/vscode-dts/vscode.proposed.terminalShellType.d.ts

```diff
@@ -7,34 +7,14 @@ declare module 'vscode' {
 
 	// https://github.com/microsoft/vscode/issues/230165
 
-	/**
-	 * Known terminal shell types.
-	 */
-	export enum TerminalShellType {
-		Sh = 1,
-		Bash = 2,
-		Fish = 3,
-		Csh = 4,
-		Ksh = 5,
-		Zsh = 6,
-		CommandPrompt = 7,
-		GitBash = 8,
-		PowerShell = 9,
-		Python = 10,
-		Julia = 11,
-		NuShell = 12,
-		Node = 13
-	}
-
 	// Part of TerminalState since the shellType can change multiple times and this comes with an event.
 	export interface TerminalState {
 		/**
-		 * The current detected shell type of the terminal. New shell types may be added in the
-		 * future in which case they will be returned as a number that is not part of
-		 * {@link TerminalShellType}.
-		 * Includes number type to prevent the breaking change when new enum members are added?
+		 * The current detected shell type of the terminal.
+		 * Standardize on shell binaries (serve as command-line interpreters). Compiled programs that read and execute command from user.
+		 * For example, bash, zsh, pwsh will be bash, zsh, pwsh accordingly.
 		 */
-		readonly shellType?: TerminalShellType | number | undefined;
+		readonly shell: string | undefined;
```

Comment:
Could you re-write this using https://github.com/microsoft/vscode/blob/6e072ba61f6119d23ae3cd5b1a70eb67b7164887/src/vscode-dts/vscode.proposed.textDocumentEncoding.d.ts#L12-L29 as inspiration?

Reply:
Got some inspiration so now it looks like: [1e60fe2](https://github.com/microsoft/vscode/pull/243274/commits/1e60fe2d456840c6b80dffcde768ae8faff0e325) but open to any further suggestion from folks!